docs/SphinxReplayDB.md

```diff
@@ -0,0 +1,123 @@
+# Sphinx Onion Routing in Lightning Network
+
+The Lightning Network uses a Sphinx-based onion message protocol to send
+messages across the Lightning Network. These messages have the property that a
+node which is part of such a message and for example forwards such an onion
+message cannot learn about the destination of this whole packet. It only knows
+the predecessor and the successor of the message. In other words, it only knows
+where the message came from and where it needs to be forwarded. This makes the
+message protocol of the Lightning Network very private. Only the sender
+(creator of the whole onion packet) knows the whole route of the packet. Also,
+the receiver has no idea from which node the message originated.
+
+This diagram illustrates how Sphinx onion routing works in the Lightning
+Network, showing the privacy properties at each hop:
+
+```ascii
+Alice (Sender)
+    |
+    | Knows Full Route: Alice → Bob → Carol → David → Eve
+    |
+    v
+Bob (Hop 1)
+    |
+    | Knows: Alice → Carol
+    |
+    v
+Carol (Hop 2)
+    |
+    | Knows: Bob → David
+    |
+    v
+David (Hop 3)
+    |
+    | Knows: Carol → Eve
+    |
+    v
+Eve (Receiver)
+    |
+    | Knows: From David
+    |
+    v
+
+Privacy Properties:
+- Each hop only knows its immediate neighbors
+- Receiver doesn't know the original sender
+- Only sender knows the complete route
+- Each hop peels one layer of the onion
+```
+
+## Privacy Properties
+
+1. **Hop Privacy**: Each intermediate node only knows:
+   - The previous hop (where the packet came from)
+   - The next hop (where to forward the packet)
+   - Cannot see the full route or final destination
+
+2. **Sender Privacy**: Only the sender (Alice) knows:
+   - The complete route
+   - All intermediate nodes
+   - The final destination
+
+3. **Receiver Privacy**: The receiver (Eve) only knows:
+   - The immediate previous hop (David)
+   - Cannot determine the original sender
+
+4. **Onion Encryption**: Each hop peels one layer of encryption, revealing only
+   the next hop's information
+
+
+The detailed mechanics are described in [BOLT 04](https://github.com/lightning/bolts/blob/master/04-onion-routing.md)
+
+
+## Replay Protection
+
+Replaying (resending) onion packets into the network can compromise the
+privacy guarantees promised by the protocol. So it is crucial for node
+participants to not forward replayed onion packets to guard the privacy of all
+network participants. Compared to the original Sphinx protocol, the Lightning
+Network has an improved replay protection in place especially when it comes to
+forwarding HTLCs, which are different from Onion Messages introduced later on
+because they lock a payment to the particular onion packet. Therefore, sending
```

Comment:
...because HTLCs lock a payment to the particular onion packet.

The above framing makes it clear IMO